File: sdk/go/leaderboard.go
package velocity

import (
	"context"
)

type LeaderboardEntry struct {
	Rank           int     `json:"rank"`
	SubmissionID   string  `json:"submission_id"`
	Team           string  `json:"team"`
	Display        string  `json:"display"`
	CompositeScore float64 `json:"composite_score"`
	LatencyScore   float64 `json:"latency_score"`
	ThroughputScore float64 `json:"throughput_score"`
	Profile        Profile `json:"profile"`
	UpdatedAtNs    int64   `json:"updated_at_ns"`
}

type LeaderboardResource struct{ c *Client }

func (r *LeaderboardResource) Top(ctx context.Context, limit int) ([]LeaderboardEntry, error) {
	var resp struct {
		Entries []LeaderboardEntry `json:"entries"`
	}
	if err := r.c.do(ctx, "GET",
		"/v1/leaderboard?limit="+intToString(limit), nil, &resp); err != nil {
		return nil, err
	}
	return resp.Entries, nil
}

File: sdk/ts/src/leaderboard.ts
/**
 * Leaderboard resource (`/v1/leaderboard`).
 *
 * Read-only; available to any authenticated submitter. Returns the
 * current top-N entries with composite/latency/throughput scores
 * already weighted by the active scoring profile.
 */

import type { VelocityClient } from './client.js'
import type { Profile } from './benchmarks.js'

export type LeaderboardEntry = {
  rank: number
  submission_id: string
  team: string
  display: string
  composite_score: number
  latency_score: number
  throughput_score: number
  profile: Profile
  updated_at_ns: number
}

export class LeaderboardResource {
  constructor(private readonly client: VelocityClient) {}

  async top(limit = 50): Promise<LeaderboardEntry[]> {
    const resp = await this.client.request<{ entries: LeaderboardEntry[] }>(
      'GET', `/v1/leaderboard?limit=${limit}`,
    )
    return resp.entries
  }
}
